src/parser/statement.js

```diff
@@ -303,35 +304,38 @@ pp.parseForStatement = function (node) {
     return this.parseFor(node, init);
   }
 
+  let usedNowKeyword = false;
+  if (this.hasPlugin("lightscript") && (!forAwait)) {
+    // idx, elem, key, or val begins a LS-enhanced loop.
+    if (
+      (this.isContextual("idx") || this.isContextual("elem") || this.isContextual("key") || this.isContextual("val"))
+    ) {
+      return this.parseEnhancedForIn(node);
+    }
+
+    // Possibly for now...in
+    usedNowKeyword = this.eat(tt._now);
```

Comment:
Okay, I think I've found the cleanup for this leaking into `parseExpression`... 

At https://github.com/lightscript/babylon-lightscript/blob/lightscript/src/parser/expression.js#L187 , try adding `|| noIn`. Then you can check `init.isNowAssign` and get rid of `usedNowKeyword = this.eat(tt._now);` above